contracts/transfer/circuits/src/gadgets/nullifier.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) DUSK NETWORK. All rights reserved.
+// Licensed under the MPL 2.0 license. See LICENSE file in the project root for details.
+
+use phoenix_core::note::Note;
+use dusk_plonk::constraint_system::ecc::scalar_mul::fixed_base::scalar_mul;
+use dusk_plonk::jubjub::{
+    Fr, AffinePoint, GENERATOR_EXTENDED, GENERATOR_NUMS_EXTENDED,
+};
+use dusk_plonk::prelude::*;
+use poseidon252::sponge::sponge::{sponge_hash_gadget, sponge_hash};
+use dusk_bls12_381::Scalar;
+use plonk_gadgets::AllocatedScalar;
+use dusk_pki::Ownable;
```

Comment:
Please check the inputs, I told you how to import everything.

Reply:
Doesn't work